src/shared/elf-util.c

```diff
@@ -266,6 +266,42 @@ static int thread_callback(Dwfl_Thread *thread, void *userdata) {
         return DWARF_CB_OK;
 }
 
+static void report_module_metadata(StackContext *c, const char *name, JsonVariant *metadata) {
+        assert(c);
+        assert(name);
+        assert(metadata);
+
+        if (!c->f)
+                return;
+
+        const char
+                *build_id = json_variant_string(json_variant_by_key(metadata, "buildId")),
+                *type = json_variant_string(json_variant_by_key(metadata, "type")),
+                *package = json_variant_string(json_variant_by_key(metadata, "name")),
+                *version = json_variant_string(json_variant_by_key(metadata, "version")),
+                *arch = json_variant_string(json_variant_by_key(metadata, "architecture"));
```

Comment:
(i wonder if we should add json_variant_by_key_as_string() that makes this one call)

Reply:
Probably.

Reply:
I'll submit that as a separate PR, since it's likely to gender some discussion.

Reply:
I have some patches prepared, but I'm waiting for this PR to go in before submitting.